### PR TITLE
W-11427407 add overwrite flag in postAction to support selection in move file operation

### PR DESCRIPTION
**Pull request by rogerwang1102**
add overwrite flag in postAction to support selection in move file operation

### CODE REVIEW

File: src/main/java/org/mule/extension/file/internal/source/PostActionGroup.java [hunk 1]
```diff
@@ -52,14 +52,24 @@ public class PostActionGroup extends AbstractPostActionGroup {
...
+  @Parameter
+  @Optional(defaultValue = "false")
+  private boolean overwrite = false;
+
 
   public PostActionGroup() {}
 
-  public PostActionGroup(boolean autoDelete, String moveToDirectory, String renameTo, boolean applyPostActionWhenFailed) {
+
+  public PostActionGroup(boolean autoDelete, String moveToDirectory, String renameTo, boolean applyPostActionWhenFailed,
```
Comment (acurci-at-mulesoft) [inline src/main/java/org/mule/extension/file/internal/source/PostActionGroup.java:66]: Hi, Roger. The fact that you had to modify the test case is a probe that you are changing the API. 
A better option would be to create another constructor with five parameters and preserve the previous one with the default value for the new attribute.  
Doing so means you don't have to change any tests (you must add new ones, but without changing the previous ones).

File: src/test/java/org/mule/extension/file/integration/PostActionGroupTestCase.java [hunk 2]
```diff
@@ -22,22 +22,22 @@ public class PostActionGroupTestCase extends AbstractMuleTestCase {
   @Test
   @Description("tests all the valid states of post action parameters")
   public void validAction() {
-    new PostActionGroup(true, null, null, true).validateSelf();
-    new PostActionGroup(true, null, null, false).validateSelf();
-    new PostActionGroup(false, "someDir", null, false).validateSelf();
-    new PostActionGroup(false, "someDir", "thisone.txt", false).validateSelf();
-    new PostActionGroup(false, null, "thisone.txt", false).validateSelf();
+    new PostActionGroup(true, null, null, true, false).validateSelf();
```
Comment (acurci-at-mulesoft) [inline src/test/java/org/mule/extension/file/integration/PostActionGroupTestCase.java:25]: You should add tests for the true value in the new parameter. 
(I have left a comment about the API change in the PostActionGroup class).

**Review by acurci-at-mulesoft — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/extension/file/internal/source/PostActionGroup.java (hunk 1 above)
Comment (rogerwang1102) [reply src/main/java/org/mule/extension/file/internal/source/PostActionGroup.java:66]: OK, got it.

File: src/test/java/org/mule/extension/file/integration/PostActionGroupTestCase.java (hunk 2 above)
Comment (rogerwang1102) [reply src/test/java/org/mule/extension/file/integration/PostActionGroupTestCase.java:25]: Yes, I will add it. 
I just wait for build to be normal.

**Review by acurci-at-mulesoft — CHANGES_REQUESTED** [review]

**Review by acurci-at-mulesoft — APPROVED** [review]

**Pull request merged**